Mathlib/Probability/Hoeffding.lean

```diff
@@ -0,0 +1,556 @@
+  have hm1 : HasFiniteIntegral (fun ω ↦ rexp (t * X ω)) μ := by
+    have b' : ∀ᵐ ω ∂μ, rexp (t * X ω) ≤ rexp (t * b) := by
+      filter_upwards [hb] with ω hb
+      using exp_le_exp.mpr (mul_le_mul_of_nonneg_left hb (le_of_lt ht))
+    have p : ∀ᵐ ω ∂μ, ‖rexp (t * X ω)‖₊ ≤ rexp (t * b) := by
+      filter_upwards [b'] with ω b'
+      rw [(by simp only [coe_nnnorm, norm_eq_abs, abs_exp] : ‖rexp (t * X ω)‖₊ = rexp (t * X ω))]
+      exact b'
+    have p'' : ∫⁻ ω, ‖rexp (t * X ω)‖₊ ∂μ ≤ ∫⁻ _, ‖rexp (t * b)‖₊ ∂μ := by
+      apply lintegral_mono_ae
+      filter_upwards [p] with ω p
+      simp only [ENNReal.coe_le_coe]
+      rw [← (by simp only [coe_nnnorm, norm_eq_abs, abs_exp] : ‖rexp (t * b)‖₊ = rexp (t * b))] at p
+      exact p
+    suffices ∫⁻ _, ↑‖rexp (t * b)‖₊ ∂μ < ⊤ from lt_of_le_of_lt p'' this
+    simp only [lintegral_const]
+    apply ENNReal.mul_lt_top ENNReal.coe_lt_top (IsFiniteMeasure.measure_univ_lt_top)
+  exact ⟨aestronglyMeasurable_iff_aemeasurable.mpr <|
+    measurable_exp.comp_aemeasurable' (hX.const_mul t), hm1⟩
+
+lemma integrable_expt_bound[IsFiniteMeasure μ] {X : Ω → ℝ} {t a b : ℝ} (hX : AEMeasurable X μ)
+    (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    Integrable (fun ω ↦ exp (t * (X ω))) μ := by
+  have ha : ∀ᵐ ω ∂μ, a ≤ X ω := h.mono fun ω h ↦ h.1
+  cases lt_trichotomy t 0 with
+  | inr ht => cases ht with
+    | inr ht => exact integrable_expt μ X t b ht hX (h.mono fun ω h ↦ h.2)
+    | inl ht => rw [ht]; simp only [zero_mul, exp_zero, integrable_const]
+  | inl ht =>
+    rw [(by ext ω; rw [(by ring : - t * (- X ω) = t * X ω)] :
+      (fun ω ↦ rexp (t * X ω)) = (fun ω ↦ rexp (- t * (- X ω))))]
+    exact integrable_expt _ _ _ _ (by linarith : -t > 0) (AEMeasurable.neg hX)
+      (by filter_upwards [ha] with ω ha using neg_le_neg ha)
+
+lemma expt_pos [IsFiniteMeasure μ] [NeZero μ] (X : Ω → ℝ) (t a b : ℝ)
+    (hX : AEMeasurable X μ) (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    0 < μ[fun ω ↦ exp (t * (X ω))] :=
+  integral_exp_pos <| integrable_expt_bound μ hX h
+
+lemma tilt_var_bound [IsProbabilityMeasure μ] (a b t : ℝ) (X : Ω → ℝ)
+    (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b)
+    (hX : AEMeasurable X μ) :
+    variance X (μ.tilted (fun ω ↦ t * X ω)) ≤ ((b - a) / 2) ^ 2 := by
+  have _ : IsProbabilityMeasure (μ.tilted fun ω ↦ t * X ω) :=
+    isProbabilityMeasure_tilted (integrable_expt_bound μ hX h)
+  exact variance_le_sq_of_bounded
+      ((tilted_absolutelyContinuous μ fun ω ↦ t * X ω) h)
+      (AEMeasurable.mono_ac hX (tilted_absolutelyContinuous μ fun ω ↦ t * X ω))
+
+theorem integrable_bounded [IsFiniteMeasure μ] (a b : ℝ)
+    (X : Ω → ℝ) (hX : AEMeasurable X μ) (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    Integrable X μ := by
+  have m1 : HasFiniteIntegral X μ := by
+    apply (hasFiniteIntegral_const (max ‖a‖ ‖b‖)).mono'
+    filter_upwards [h.mono fun ω h ↦ h.1, h.mono fun ω h ↦ h.2] with ω using abs_le_max_abs_abs
+  exact ⟨aestronglyMeasurable_iff_aemeasurable.mpr hX, m1⟩
```

Comment:
You already proved [MeasureTheory.memℒp_of_bounded](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Function/LpSeminorm/Basic.html#MeasureTheory.mem%E2%84%92p_of_bounded) so you could use first [MeasureTheory.memℒp_one_iff_integrable](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Function/L1Space.html#MeasureTheory.mem%E2%84%92p_one_iff_integrable) and then the Lp result.